src/main/java/io/github/townyadvanced/townycombat/utils/TownyCombatMovementUtil.java

```diff
@@ -1,63 +1,174 @@
+        //Apply adjustments to player walk speed
+        adjustPlayerWalkSpeed(player);
+        //Apply adjustments to mount walk speed
+        adjustMountWalkSpeed(player);
+    }
+
+    private static void adjustPlayerWalkSpeed(Player player) {
         //Calculate generic speed adjustment
         double genericSpeedAdjustmentPercentage = TownyCombatSettings.getGenericInfantrySpeedAdjustmentPercentage();
 
-        //Calculate armour speed adjustment
-        Map<Material, Double> materialSlowPercentageMap = TownyCombatSettings.getMaterialSlowPercentageMap();
-        double armourSpeedAdjustmentPercentage = 0;
-        Double slowPercentage;
+        //Calculate slow due to encumbrance
+        Map<Material, Double> materialEncumbrancePercentageMap = TownyCombatSettings.getMaterialEncumbrancePercentageMap();
+        double totalEncumbrancePercentage = 0;
+        Double itemEncumbrancePercentage;
         for(ItemStack itemStack: player.getInventory().getContents()) {
             if(itemStack != null) {
-                slowPercentage = materialSlowPercentageMap.get(itemStack.getType());
-                if(slowPercentage != null) {
-                    armourSpeedAdjustmentPercentage -= slowPercentage;
+                itemEncumbrancePercentage = materialEncumbrancePercentageMap.get(itemStack.getType());
+                if(itemEncumbrancePercentage != null) {
+                    totalEncumbrancePercentage += itemEncumbrancePercentage;
                 }
             }
         }      
-        infantryArmourSpeedAdjustmentPercentages.put(player, armourSpeedAdjustmentPercentage);
+        playerEncumbrancePercentageMap.put(player, totalEncumbrancePercentage);
 
-        //Apply all speed adjustments
+        //Calculate total speed adjustment
         double recalculatedSpeed = 
-            VANILLA_PLAYER_GENERIC_MOVEMENT_SPEED + (VANILLA_PLAYER_GENERIC_MOVEMENT_SPEED * ((genericSpeedAdjustmentPercentage + armourSpeedAdjustmentPercentage) / 100));
+            VANILLA_PLAYER_MOVEMENT_SPEED + (VANILLA_PLAYER_MOVEMENT_SPEED * ((genericSpeedAdjustmentPercentage - totalEncumbrancePercentage) / 100));
+        //Sanitize
+        recalculatedSpeed = Math.max(recalculatedSpeed, 0.05);
+        recalculatedSpeed = Math.min(recalculatedSpeed, 1);
+       //Apply
+        player.getAttribute(Attribute.GENERIC_MOVEMENT_SPEED).setBaseValue(recalculatedSpeed);
+    }
+    
+    private static void adjustMountWalkSpeed(Player player) {
+        if(!player.isInsideVehicle() || !(player.getVehicle() instanceof AbstractHorse))
+            return;
+        AbstractHorse mount = (AbstractHorse)player.getVehicle();
+        AnimalTamer owner = mount.getOwner();
+        if(owner == null) 
+            return;
+        
+        //Get Base Walk Speed
+        Double baseWalkSpeed;
+        UUID ownerUUID = mount.getOwner().getUniqueId();
+        
+        System.out.println("Mount owner " + ownerUUID);
```

Comment:
sysout will make paper unhappy, I think I count 5 of them in total in this class.

Reply:
Ah yes!.
Oopsie I forgot to remove those, they were just for dev purposes